src/Nethermind/Nethermind.Core/Crypto/PublicKey.cs

```diff
@@ -136,3 +140,39 @@ public string ToShortString()
         }
     }
 }
+
+namespace Nethermind.Serialization.Json
+{
+    using Nethermind.Core.Crypto;
+
+    public class PublicKeyConverter : JsonConverter<PublicKey>
+    {
+        public override PublicKey? Read(
+            ref Utf8JsonReader reader,
+            Type typeToConvert,
+            JsonSerializerOptions options)
+        {
+            byte[]? bytes = ByteArrayConverter.Convert(ref reader);
+            if (bytes is null)
+            {
+                return null;
+            }
+            if (bytes.Length < 64)
+            {
+                var newArray = new byte[64];
+                bytes.AsSpan().CopyTo(newArray.AsSpan(64 - bytes.Length));
+                bytes = newArray;
+            }
```

Comment:
Other types don't have that safety measure? 
Maybe this should be pre-allocated and passed to `ByteArrayConverter.Convert` ?